### PR TITLE
[Feature] Lap(Flag) of TimeInterval added

### LINKED ISSUE #406: Feature: Adding Lap(Flag) in stopwatch  to track intervals
### Description

Adding a lap feature to an alarm clock enhances its functionality by allowing users to track time intervals, typically used during activities like running, exercising, or timing laps in sports events. The lap feature enables users to mark specific points in time, creating segments for easier monitoring and analysis.

Display lap times in a separate section or list within the interface.

When the user initiates a lap, record the current time and display it as a lap time.

Allow users to view lap times sequentially as they mark additional laps.

### Screenshots

WhatsApp Image 2024-01-30 at 6 10 3
Android Large - 1

### PR DESCRIPTION
**Pull request by codesage01**
### Description
New Lap(Flag) feature added,
When user start the stopwatch then reset button changes to flag button,
After pressing flag user get result of Timeinterval at the time of press.
When user pause the stopwatch then flag button converted to reset button,
and user if want then reset the stopwatch by pressing.

 ### Fixes #406

### Screenshots

https://github.com/CCExtractor/ultimate_alarm_clock/assets/149355139/613e77f1-536b-49f2-992d-6288ae9322e9

## Checklist

- [ ] Tests have been added or updated to cover the changes
- [ ] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [ ] All tests are passing

@MarkisDev, could you please review this PR

### CODE REVIEW

File: lib/app/modules/alarmRing/views/alarm_ring_view.dart
Comment (MarkisDev) [inline lib/app/modules/alarmRing/views/alarm_ring_view.dart:1]: View file cannot contain any business logic, please keep it only for the UI.

File: lib/app/modules/alarmRing/controllers/alarm_ring_controller.dart [hunk 1]
```diff
@@ -52,7 +52,7 @@ class AlarmControlController extends GetxController {
     return latestAlarm;
   }
 
-  getNextAlarm() async {
+  getAlarm() async {
```
Comment (MarkisDev) [inline lib/app/modules/alarmRing/controllers/alarm_ring_controller.dart:55]: Why was this function renamed?

**Review by MarkisDev — CHANGES_REQUESTED** [review]

File: lib/app/modules/alarmRing/controllers/alarm_ring_controller.dart (hunk 1 above)
Comment (codesage01) [reply lib/app/modules/alarmRing/controllers/alarm_ring_controller.dart:55]: sorry, i replaced next unfortunately. i will solved all the problem make a fresh commit. Thanks

File: lib/app/modules/stopwatch/views/stopwatch_view.dart [hunk 2]
```diff
@@ -21,120 +21,195 @@ class StopwatchView extends GetView<StopwatchController> {
...
+                      ? kLightPrimaryTextColor.withOpacity(0.8)
+                      : kprimaryTextColor.withOpacity(0.8),
                 ),
               ),
             ],
           ),
-        ],
+        ),
       ),
-      endDrawer: buildEndDrawer(context)
```
Comment (MarkisDev) [inline lib/app/modules/stopwatch/views/stopwatch_view.dart:137]: Why was this removed?

Comment (MarkisDev) [reply lib/app/modules/stopwatch/views/stopwatch_view.dart:137]: We're using the buildEndDrawer to keep the drawers consistent.

**Review by MarkisDev — CHANGES_REQUESTED** [review]

File: lib/app/modules/stopwatch/views/stopwatch_view.dart (hunk 2 above)
Comment (codesage01) [reply lib/app/modules/stopwatch/views/stopwatch_view.dart:137]: i used obx for EndDrawer in line no.89 to keep drawers  consistent.

Comment (MarkisDev) [reply lib/app/modules/stopwatch/views/stopwatch_view.dart:137]: We have a function called `buildEndDrawer` in `utils/end_drawer` which returns the endDrawer. We can reuse this function in order to make any change to the endDrawer reflect in all three pages.

**Review by MarkisDev — CHANGES_REQUESTED** [review]
